### PR TITLE
Clean up docs Makefile

### PR DESCRIPTION
**Pull request by eracah**
This PR:
* add `-q` flag, so only warnings and failures are shown
* add `clean` command that will automatically remove `source/api_reference`
* adjust docs accordingly
* adjust jenkins accordingly

### CODE REVIEW

**Review by mvpatel2000 — APPROVED** [review]
lgtm

File: docs/Makefile [hunk 1]
```diff
@@ -3,7 +3,7 @@
 
 # You can set these variables from the command line, and also
 # from the environment for the first two.
-SPHINXOPTS    ?= -W --keep-going
+SPHINXOPTS    ?= -W --keep-going -Q
```
Comment (hanlint) [inline docs/Makefile:6]: `-Q` would also suppress warnings, which we consider errors (see `-W` flag). In my testing on this branch, if I introduce a warning, I get:
```
$ make html
/Users/hanlintang/composer-fork/docs/source/conf.py:108: UserWarning: The git repo is dirty. The commit sha for source code links will be incorrect.
  warnings.warn(warning_msg)
Generating rst file /Users/hanlintang/composer-fork/docs/source/api_reference/composer.rst for module: composer
Generating rst file /Users/hanlintang/composer-fork/docs/source/api_reference/composer.algorithms.rst for module: composer.algorithms
Generating rst file /Users/hanlintang/composer-fork/docs/source/api_reference/composer.callbacks.rst for module: composer.callbacks
Generating rst file /Users/hanlintang/composer-fork/docs/source/api_reference/composer.core.rst for module: composer.core
Generating rst file /Users/hanlintang/composer-fork/docs/source/api_reference/composer.core.types.rst for module: composer.core.types
Generating rst file /Users/hanlintang/composer-fork/docs/source/api_reference/composer.datasets.rst for module: composer.datasets
Generating rst file /Users/hanlintang/composer-fork/docs/source/api_reference/composer.functional.rst for module: composer.functional
Generating rst file /Users/hanlintang/composer-fork/docs/source/api_reference/composer.functional.rst for module: composer.functional
Generating rst file /Users/hanlintang/composer-fork/docs/source/api_reference/composer.loggers.rst for module: composer.loggers
Generating rst file /Users/hanlintang/composer-fork/docs/source/api_reference/composer.loss.rst for module: composer.loss
Generating rst file /Users/hanlintang/composer-fork/docs/source/api_reference/composer.metrics.rst for module: composer.metrics
Generating rst file /Users/hanlintang/composer-fork/docs/source/api_reference/composer.models.rst for module: composer.models
[ommited: several "generating.. lines]
make: *** [html] Error 1
```

which is missing the critical error:
```
/Users/hanlintang/composer-fork/docs/source/trainer/using_the_trainer.rst:462: WARNING: unknown document: /abc
```

Could we revert the `-Q`?

**Review by hanlint — CHANGES_REQUESTED** [review]

File: docs/Makefile (hunk 1 above)
Comment (eracah) [reply docs/Makefile:6]: Oh good call. It actually should be `-q`. See https://www.sphinx-doc.org/en/master/man/sphinx-build.html#cmdoption-sphinx-build-q

Comment (eracah) [reply docs/Makefile:6]: With `-q`, we will get the proper warning:
```
(ec-venv) Evans-MacBook-Pro:docs evanracah$ make html
/Users/evanracah/code/evan-composer/docs/source/conf.py:108: UserWarning: The git repo is dirty. The commit sha for source code links will be incorrect.
  warnings.warn(warning_msg)
Generating rst file /Users/evanracah/code/evan-composer/docs/source/api_reference/composer.rst for module: composer
Generating rst file /Users/evanracah/code/evan-composer/docs/source/api_reference/composer.algorithms.rst for module: composer.algorithms
Generating rst file /Users/evanracah/code/evan-composer/docs/source/api_reference/composer.callbacks.rst for module: composer.callbacks
Generating rst file /Users/evanracah/code/evan-composer/docs/source/api_reference/composer.core.rst for module: composer.core
Generating rst file /Users/evanracah/code/evan-composer/docs/source/api_reference/composer.core.types.rst for module: composer.core.types
Generating rst file /Users/evanracah/code/evan-composer/docs/source/api_reference/composer.datasets.rst for module: composer.datasets
Generating rst file /Users/evanracah/code/evan-composer/docs/source/api_reference/composer.functional.rst for module: composer.functional
Generating rst file /Users/evanracah/code/evan-composer/docs/source/api_reference/composer.functional.rst for module: composer.functional
Generating rst file /Users/evanracah/code/evan-composer/docs/source/api_reference/composer.loggers.rst for module: composer.loggers
Generating rst file /Users/evanracah/code/evan-composer/docs/source/api_reference/composer.loss.rst for module: composer.loss
Generating rst file /Users/evanracah/code/evan-composer/docs/source/api_reference/composer.metrics.rst for module: composer.metrics
Generating rst file /Users/evanracah/code/evan-composer/docs/source/api_reference/composer.models.rst for module: composer.models
Generating rst file /Users/evanracah/code/evan-composer/docs/source/api_reference/composer.optim.rst for module: composer.optim
Generating rst file /Users/evanracah/code/evan-composer/docs/source/api_reference/composer.profiler.rst for module: composer.profiler
Generating rst file /Users/evanracah/code/evan-composer/docs/source/api_reference/composer.utils.rst for module: composer.utils
Generating rst file /Users/evanracah/code/evan-composer/docs/source/api_reference/composer.utils.dist.rst for module: composer.utils.dist
Generating rst file /Users/evanracah/code/evan-composer/docs/source/api_reference/composer.utils.reproducibility.rst for module: composer.utils.reproducibility
/Users/evanracah/code/evan-composer/docs/source/trainer/using_the_trainer.rst:462: WARNING: unknown document: /abc
make: *** [html] Error 1
```

Comment (eracah) [reply docs/Makefile]: fixed

Comment (eracah) [reply docs/Makefile]: Ok I added `-q` just to `make doctest`, so `make html` can still be a little bit verbose.

**Review by hanlint — APPROVED** [review]
LGTM

**Pull request merged**